packages/ui-components/src/ExpandableErrorDialog.tsx

```diff
@@ -31,58 +31,53 @@ interface IErrorDialogProps {
   default_msg: string;
 }
 
-export class ExpandableErrorDialog extends React.Component<
-  IErrorDialogProps,
-  any
-> {
-  dialogNode: HTMLDivElement;
-  collapsedDimensions: number[];
+export const ExpandableErrorDialog: React.FC<IErrorDialogProps> = ({
+  reason,
+  message,
+  timestamp,
+  traceback,
+  default_msg
+}) => {
+  const [collapsedSize, setCollapsedSize] = React.useState(null);
 
-  constructor(props: any) {
-    super(props);
-  }
+  const handleUpdateDialogSize = React.useCallback(
+    (expanded: boolean): void => {
+      const dialogNode: HTMLDivElement = document.querySelector(
+        '.' + JP_DIALOG_CONTENT
+      );
+      const width = dialogNode.clientWidth;
+      const height = dialogNode.clientHeight;
 
-  updateDialogSize(expanded: boolean): void {
-    if (!this.dialogNode) {
-      this.dialogNode = document.querySelector('.' + JP_DIALOG_CONTENT);
-    }
+      if (
+        expanded &&
+        (width < ERROR_DIALOG_WIDTH || height < ERROR_DIALOG_HEIGHT)
+      ) {
+        setCollapsedSize({ width, height });
+        dialogNode.style.width = Math.max(width, ERROR_DIALOG_WIDTH) + 'px';
+        dialogNode.style.height = Math.max(height, ERROR_DIALOG_HEIGHT) + 'px';
+      } else if (!expanded && collapsedSize) {
+        dialogNode.style.width = collapsedSize.width + 'px';
+        dialogNode.style.height = collapsedSize.height + 'px';
+      }
+    },
```

Comment:
We are manually mutating the DOM here, which is normally considered a side effect, so this should go into a `useEffect` hook

Reply:
```tsx
const [expanded, setExpanded] = useState(null);

// we can use a ref, because we don't need to rerender the component if its value changes
// this could also be initialized to undefined or null if that is something that needs to be checked
const collapsedSizeRef = useRef({ width: 0, height: 0 });

useEffect(() => {
    const dialogNode = document.querySelector(JP_DIALOG_CONTENT);
    if (expanded) {
        collapsedSizeRef.current = { width: dialogNode.clientWidth,  height: dialogNode.clientHeight}
        dialogNode.style.width = Math.max(width, ERROR_DIALOG_WIDTH) + 'px';
        dialogNode.style.height = Math.max(height, ERROR_DIALOG_HEIGHT) + 'px';
    } else {
        dialogNode.style.width = collapsedSize.current.width + 'px';
        dialogNode.style.height = collapsedSize.current.height + 'px';
    }
}, [expanded])

// this doesn't really need to be a callback hook
const handleSizeChange = (expanded: boolean) => {
    setExpanded(expanded);
}
```

Reply:
Just to document here the discussions around this topic, the proposed solution changes the DOM tree and therefore the original dialog resize behaviour.
Another solution we've tried was using CSS selectors only, but it didn't work in some cases.
This is a tricky case as we are manipulating and tracking the state of an element external to this component, maybe that's why using effects here doesn't really behave as expected. 
Therefore it was decided to keep it as is.